ee/tabby-webserver/src/repositories/resolve.rs

```diff
@@ -1,88 +1,65 @@
     Json,
 };
 use hyper::Body;
+use kv::{Bucket, Config, Store};
 use serde::{Deserialize, Serialize};
 use tabby_common::{config::RepositoryConfig, SourceFile, Tag};
 use tower::ServiceExt;
 use tower_http::services::ServeDir;
-use tracing::{debug, error, warn};
-
-use crate::{
-    cron::{CronEvents, StartListener},
-    schema::repository::RepositoryService,
-};
 
 pub struct RepositoryCache {
-    repository_lookup: RwLock<HashMap<RepositoryKey, RepositoryMeta>>,
-    service: Arc<dyn RepositoryService>,
+    cache: Store,
 }
 
 impl std::fmt::Debug for RepositoryCache {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        f.debug_struct("RepositoryCache")
-            .field("repository_lookup", &self.repository_lookup)
-            .finish()
+        f.debug_struct("RepositoryCache").finish()
     }
 }
+type RepositoryBucket<'a> = Bucket<'a, String, kv::Json<SourceFile>>;
 
 impl RepositoryCache {
-    pub async fn new_initialized(
-        service: Arc<dyn RepositoryService>,
-        events: &CronEvents,
-    ) -> Arc<RepositoryCache> {
-        let cache = RepositoryCache {
-            repository_lookup: Default::default(),
-            service,
-        };
-        if let Err(e) = cache.reload().await {
-            error!("Failed to load repositories: {e}");
-        };
-        let cache = Arc::new(cache);
-        cache.start_reload_listener(events);
-        cache
+    pub fn new() -> Result<Self> {
+        let config = Config::new(tabby_common::path::repository_meta_db());
+        let store = Store::new(config)?;
+        Ok(RepositoryCache { cache: store })
+    }
+
+    fn bucket(&self) -> Result<RepositoryBucket> {
+        Ok(self.cache.bucket(Some("repositories"))?)
     }
 
-    async fn reload(&self) -> Result<()> {
-        let new_repositories = self
-            .service
-            .list_repositories(None, None, None, None)
-            .await?
-            .into_iter()
-            .map(|repository| RepositoryConfig::new_named(repository.name, repository.git_url))
-            .collect();
-        let mut repository_lookup = self.repository_lookup.write().unwrap();
-        debug!("Reloading repositoriy metadata...");
-        *repository_lookup = load_meta(new_repositories);
+    pub fn clear(&self) -> Result<()> {
+        self.bucket()?.clear()?;
         Ok(())
     }
 
-    fn start_reload_listener(self: &Arc<Self>, events: &CronEvents) {
-        let clone = self.clone();
-        events.scheduler_job_succeeded.start_listener(move |_| {
-            let clone = clone.clone();
-            async move {
-                if let Err(e) = clone.reload().await {
-                    warn!("Error when reloading repository cache: {e}");
-                };
-            }
-        });
+    pub fn add_repository_meta(&self, file: SourceFile) -> Result<()> {
+        let key = format!("{}:{}", file.repository_name, file.filepath);
+        self.bucket()?.set(&key, &kv::Json(file))?;
```

Comment:
SourceFile content shouldn't be inside of cache. Any reason you don't store `RepositoryMeta` instead?